### PR TITLE
cluster singleton improvements, #20942

### PR DESCRIPTION
**Pull request by patriknw**
see the 3 commits that are improving different things

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #20942

**Comment by patriknw** [comment]
rebased, mima conflict

**Comment by patriknw** [comment]
@johanandren @drewhk @2m a few eyes on this would be good

File: akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala [hunk 1]
```diff
@@ -504,6 +504,12 @@ class ClusterSingletonManager(
       addRemoved(m.uniqueAddress)
       // transition when OldestChanged
       stay using YoungerData(None)
+
+    case Event(HandOverToMe, _) ⇒
+      // this node was probably quickly restarted with same hostname:port,
+      // confirm that the old singleton instance has been stopped
+      sender() ! HandOverDone
+      stay
```
Comment (johanandren) [inline akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala]: 👍

**Comment by johanandren** [comment]
Wouldn't it be useful if the log entries wrote the uniqueAddress rather than only the address so that it's possible to se in logs when debugging that entries are about different incarnations?

**Comment by johanandren** [comment]
LGTM!

**Comment by 2m** [comment]
LGTM

**Comment by johanandren** [comment]
When I think about that comment about the logging it might not be very interesting after the uniqueAddress changes as the mixing-up incarnations kind of problem will not be happening any more?

**Comment by patriknw** [comment]
yeah, it might be adding too much noise to the logs (for normal users)

**Comment by patriknw** [comment]
failure was unrelated ConnectionPoolSpec, issue https://github.com/akka/akka/issues/21231 created

**Comment by patriknw** [comment]
PLS BUILD

**Comment by drewhk** [comment]
LGTM

**Pull request merged**